test/testFunction.php

```diff
@@ -0,0 +1,28 @@
+<?php
+require_once '../function.php';
+
+use PHPUnit\Framework\TestCase;
+
+class FunctionTest extends TestCase
+{
+    public function testGenerateRequestShouldReturnEverythingIfNoParameter()
+    {
+        //setup
+        $result = generateRequest();
+        $expectedResult = 'SELECT `item`, `category`, `price`, `remaining` FROM `grocery_item`';
+
+        //assertion
+        $this->assertEquals($result, $expectedResult);
+    }
+
+    public function testGenerateRequestShouldReturnWhereCategoryWithParam()
+    {
+        //setup
+        $filter = 'pantry';
+        $result = generateRequest($filter);
+        $expectedResult = 'SELECT `item`, `category`, `price`, `remaining` FROM `grocery_item` WHERE `category` = "' . $filter.  '";';
+
+        //assertion
+        $this->assertEquals($result, $expectedResult);
```

Comment:
Could maybe do with failure and malformed data unit tests as well? type error etc.?

Reply:
did malformed unit test! thanks